src/import-data/ImportData.test.ts

```diff
@@ -0,0 +1,448 @@
+
+      const importUrl = 'https://example.com/path/to/file.json';
+      setup({
+        mockAjax: {
+          Workspaces: {
+            workspace: getWorkspaceApi,
+          },
+        },
+        queryParams: {
+          format: 'entitiesJson',
+          url: importUrl,
+        },
+      });
+
+      // Act
+      await importIntoExistingWorkspace(user, defaultGoogleWorkspace.workspace.name);
+
+      // Assert
+      expect(getWorkspaceApi).toHaveBeenCalledWith(
+        defaultGoogleWorkspace.workspace.namespace,
+        defaultGoogleWorkspace.workspace.name
+      );
+
+      expect(importJSON).toHaveBeenCalledWith(importUrl);
+    });
+  });
+
+  describe('TDR imports', () => {
+    describe('TDR exports', () => {
+      const queryParams = {
+        format: 'tdrexport',
+        snapshotId: '00001111-2222-3333-aaaa-bbbbccccdddd',
+        snapshotName: 'test-snapshot',
+        tdrmanifest: 'https://example.com/path/to/manifest.json',
+        tdrSyncPermissions: 'true',
+        url: 'https://data.terra.bio',
+      };
+
+      it('imports TDR exports into Google workspaces', async () => {
+        // Arrange
+        const user = userEvent.setup();
+
+        const importJob = jest.fn().mockResolvedValue({ jobId: 'new-job' });
+        const getWorkspaceApi = jest.fn().mockReturnValue({
+          importJob,
+        });
+
+        setup({
+          mockAjax: {
+            Workspaces: {
+              workspace: getWorkspaceApi,
+            },
+          },
+          queryParams,
+        });
+
+        // Act
+        await importIntoExistingWorkspace(user, defaultGoogleWorkspace.workspace.name);
+
+        // Assert
+        expect(getWorkspaceApi).toHaveBeenCalledWith(
+          defaultGoogleWorkspace.workspace.namespace,
+          defaultGoogleWorkspace.workspace.name
+        );
+
+        expect(importJob).toHaveBeenCalledWith(queryParams.tdrmanifest, 'tdrexport', { tdrSyncPermissions: true });
+      });
+
+      it('copies snapshot into Azure workspaces', async () => {
+        // Arrange
+        const user = userEvent.setup();
+
+        const importTdr = jest.fn().mockResolvedValue(undefined);
+
+        const wdsProxyUrl = 'https://proxyurl';
+        setup({
+          mockAjax: {
+            Apps: {
+              listAppsV2: jest.fn().mockResolvedValue([
+                {
+                  appType: 'WDS',
+                  appName: `wds-${defaultAzureWorkspace.workspace.workspaceId}`,
+                  status: 'RUNNING',
+                  proxyUrls: { wds: wdsProxyUrl },
+                  workspaceId: defaultAzureWorkspace.workspace.workspaceId,
+                },
+              ]),
+            },
+            WorkspaceData: {
+              importTdr,
+            },
+          },
+          queryParams,
+        });
+
+        // Act
+        await importIntoExistingWorkspace(user, defaultAzureWorkspace.workspace.name);
+
+        // Assert
+        expect(importTdr).toHaveBeenCalledWith(
```

Comment:
It seems kind of odd that we totally ignore the TDR manifest here and instead only use the snapshot ID. This means that a different subset of the query parameters that TDR passes are actually required for imports into GCP vs Azure workspaces.

Reply:
I don't know about this part of the code, I think @calypsomatic was the most recent one to work on this, when azure support was added.

Reply:
Ah, probably I just grabbed only the id because the stub importSnapshot method in wds was just looking for a snapshot id; when that method gets built out correctly presumably it would take in the whole thing.  But as it stands, it's expecting to receive just a UUID

Reply:
That makes sense. I forgot that WDS' snapshot import is currently kind of a stub.